internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -0,0 +1,501 @@
+		AccessId:    plan.AccessKeyID.ValueString(),
+		AccessToken: plan.SecretAccessKey.ValueString(),
+		ProjectID:   plan.ProjectID.ValueString(),
+		Integration: infisical.IntegrationAuthTypeAwsParameterStore,
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create integration auth",
+			err.Error(),
+		)
+		return
+	}
+
+	parsedOptions, hasError := extractOptionsParameterStore(ctx, &resp.Diagnostics, plan.Options)
+
+	if hasError {
+		return
+	}
+
+	// Convert metadata to map[string]interface{} if needed
+	metadataMap := map[string]interface{}{
+		"secretAWSTag":        parsedOptions.SecretAWSTag,
+		"shouldDisableDelete": parsedOptions.ShouldDisableDelete,
+	}
+
+	// Create the integration
+	integration, err := r.client.CreateIntegration(infisical.CreateIntegrationRequest{
+		IntegrationAuthID: auth.IntegrationAuth.ID,
+		Region:            plan.AWSRegion.ValueString(),
+		Metadata:          metadataMap,
+		SecretPath:        plan.SecretPath.ValueString(),
+		Path:              plan.AWSPath.ValueString(),
+		SourceEnvironment: plan.Environment.ValueString(),
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create integration",
+			err.Error(),
+		)
+		return
+	}
+
+	plan.IntegrationAuthID = types.StringValue(auth.IntegrationAuth.ID)
+	plan.IntegrationID = types.StringValue(integration.Integration.ID)
+
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *IntegrationAWSParameterStoreResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	if !r.client.Config.IsMachineIdentityAuth {
+		resp.Diagnostics.AddError(
+			"Unable to create integration",
+			"Only Machine Identity authentication is supported for this operation",
+		)
+		return
+	}
+
+	// Get current state
+	var state IntegrationAWSParameterStoreResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	integration, err := r.client.GetIntegration(infisical.GetIntegrationRequest{
+		ID: state.IntegrationID.ValueString(),
+	})
+
+	if err != nil {
+		if err == infisical.ErrNotFound {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to get integration",
+				err.Error(),
+			)
+		}
+		return
+	}
+
+	var planOptions AwsParameterStoreOptions
+
+	if !state.Options.IsNull() {
+		diags := state.Options.As(ctx, &planOptions, basetypes.ObjectAsOptions{})
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	updateNeeded := false
+
+	if integration.Integration.Metadata.ShouldDisableDelete != planOptions.ShouldDisableDelete {
+		planOptions.ShouldDisableDelete = integration.Integration.Metadata.ShouldDisableDelete
+		updateNeeded = true
+	}
```

Comment:
curious, why can't we just set the plan values directly to the one returned by the `integration` object?